UOP1_Project/Assets/Scripts/Editor/GameSceneEditor.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.SceneManagement;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+[CustomEditor(typeof(GameScene), true)]
+public class GameSceneEditor : Editor
+{
+    #region UI_Warnings
+
+    private const string noScenesWarning = "There are no scenes set for this level yet! Add a new scene with the dropdown above";
+    
+    #endregion
+
+    #region GUI_Styles
+
+    private GUIStyle headerLabelStyle;
+
+    #endregion
+    
+    // it holds a list of properties to hide from basic inspector
+    private static readonly string[] ExcludedProperties = {"m_Script", "sceneName"};
+    
+    private string[] sceneList;
+    private GameScene gameSceneTarget;
+
+    private void OnEnable()
+    {
+        gameSceneTarget = target as GameScene;
+        PopulateScenePicker();
+        InitializeGuiStyles();
+    }
+
+    public override void OnInspectorGUI()
+    {
+        EditorGUILayout.LabelField("Scene information", headerLabelStyle);
+        EditorGUILayout.Space();
+        DrawScenePicker();
+        DrawPropertiesExcluding(serializedObject, ExcludedProperties);
+    }
+
+    private void DrawScenePicker()
+    {
+        var sceneName = gameSceneTarget.sceneName;
+        EditorGUI.BeginChangeCheck();
+        var selectedScene = sceneList.ToList().IndexOf(sceneName);
+        selectedScene = EditorGUILayout.Popup("Scene", selectedScene, sceneList);
+        if (selectedScene > -1)
+        {
+            if (EditorGUI.EndChangeCheck())
+            {
+                Undo.RecordObject(target, "Changed selected scene");
+                gameSceneTarget.sceneName = sceneList[selectedScene];
+                MarkAllDirty();
+            }
+        }
+        else
+        {
+            EditorGUILayout.HelpBox(noScenesWarning, MessageType.Warning);
```

Comment:
There should be a call to `EditorGUI.EndChangeCheck();` here to avoid unbalanced calls when there is no scene selected yet

Reply:
Fixed it